crypto/rand/rand_unix.c

```diff
@@ -325,6 +329,90 @@ static struct random_device {
 } random_devices[OSSL_NELEM(random_device_paths)];
 static int keep_random_devices_open = 1;
 
+#   if defined(DEVRANDOM_WAIT)
+static void *shm_addr;
+
+#    if !defined(FIPS_MODE)
+static void cleanup_shm(void)
+{
+    shmdt(shm_addr);
+}
+#    endif
+#   endif
+
+/*
+ * Ensure that the system randomness source has been adequately seeded.
+ * This is done by having the first start of libcrypto, read one bytes from
+ * /dev/random.  Subsequent starts of the library and later reseedings do not
+ * need to do this.
+ */
+#   if defined(__linux)
+static int wait_random_seeded(void)
+{
+    static int seeded = 0;
+    static const int kernel_version[] = { DEVRANDOM_SAFE_KERNEL };
+    int kernel[2];
+    int shm_id, fd = -1, r;
+    char c, *p;
+    fd_set fds;
+    struct utsname un;
+
+    if (!seeded && OPENSSL_RAND_SEED_DEVRANDOM_SHM_ID >= 0) {
+        /* See if anthing has created the global seeded indication */
+        if ((shm_id = shmget(OPENSSL_RAND_SEED_DEVRANDOM_SHM_ID, 1, 0)) == -1) {
+            /* Check the kernel's version and fail if it is too recent */
```

Comment:
you could add to the comment that getrandom syscall ought to have succeeded so something went wrong

Reply:
Will do.